Framework/Algorithms/inc/MantidAlgorithms/GeometryInfo.h

```diff
@@ -0,0 +1,39 @@
+#ifndef MANTID_ALGORITHMS_GEOMETRYINFO_H_
+#define MANTID_ALGORITHMS_GEOMETRYINFO_H_
+
+#include "MantidAlgorithms/BasicInstrumentInfo.h"
+
+namespace Mantid {
+
+namespace Geometry {
+class Instrument;
+class IDetector;
+}
+
+namespace API {
+class ISpectrum;
+}
+
+namespace Algorithms {
+
+class MANTID_ALGORITHMS_DLL GeometryInfo {
```

Comment:
I think I might rename it into `GeometryInfoFactory` (and make it a factory for `GeometryInfo`). How about that?